src/circuit/circuit_frame.rs

```diff
@@ -3305,155 +3353,61 @@ fn apply_continuation<F: LurkField, CS: ConstraintSystem<F>>(
-            &mut cs.namespace(|| "comp val hash a neg and b pos"),
-            op2.tag(),
-            &[
-                CaseClause {
-                    key: Op2::Less.as_field(),
-                    value: &g.t_ptr.hash(),
-                },
-                CaseClause {
-                    key: Op2::LessEqual.as_field(),
-                    value: &g.t_ptr.hash(),
-                },
-                CaseClause {
-                    key: Op2::Greater.as_field(),
-                    value: &g.nil_ptr.hash(),
-                },
-                CaseClause {
-                    key: Op2::GreaterEqual.as_field(),
-                    value: &g.nil_ptr.hash(),
-                },
-            ],
             &g.default_num,
-        )?;
-        let comp_val_a_neg_and_b_pos = AllocatedPtr::from_parts(
-            &comp_val_tag_a_neg_and_b_pos,
-            &comp_val_hash_a_neg_and_b_pos
-        );
-
-        let comp_val_tag_a_pos_and_b_neg = case(
-            &mut cs.namespace(|| "comp val tag a pos and b neg"),
-            op2.tag(),
-            &[
-                CaseClause {
-                    key: Op2::Less.as_field(),
-                    value: &g.nil_ptr.tag(),
-                },
-                CaseClause {
-                    key: Op2::LessEqual.as_field(),
-                    value: &g.nil_ptr.tag(),
-                },
-                CaseClause {
-                    key: Op2::Greater.as_field(),
-                    value: &g.t_ptr.tag(),
-                },
-                CaseClause {
-                    key: Op2::GreaterEqual.as_field(),
-                    value: &g.t_ptr.tag(),
-                },
-            ],
             &g.default_num,
-        )?;
-        let comp_val_hash_a_pos_and_b_neg = case(
-            &mut cs.namespace(|| "comp val hash a pos and b neg"),
-            op2.tag(),
-            &[
-                CaseClause {
-                    key: Op2::Less.as_field(),
-                    value: &g.nil_ptr.hash(),
-                },
-                CaseClause {
-                    key: Op2::LessEqual.as_field(),
-                    value: &g.nil_ptr.hash(),
-                },
-                CaseClause {
-                    key: Op2::Greater.as_field(),
-                    value: &g.t_ptr.hash(),
-                },
-                CaseClause {
-                    key: Op2::GreaterEqual.as_field(),
-                    value: &g.t_ptr.hash(),
-                },
-            ],
             &g.default_num,
+            &g.false_num,
+        ];
+
+        let comp_clauses = [
+            &comp_results.same_sign_tag[..],
+            &comp_results.same_sign_hash[..],
+            &comp_results.a_neg_and_b_pos_tag[..],
+            &comp_results.a_neg_and_b_pos_hash[..],
+            &comp_results.a_pos_and_b_neg_tag[..],
+            &comp_results.a_pos_and_b_neg_hash[..],
+        ];
+
+        let comp_results = multi_case(
```

Comment:
Having this be mutable and reassigning to it is pretty confusing. What is it gained by doing so?

Reply:
Renamed the second variable.